### PR TITLE
Redux in react components

### PR DESCRIPTION
**Pull request by Diegogagan2587**
### Redux in react components

# Implemented Changes:
- Implement Connection between store and Components
- Books are now being displayed from store.js
- Include a hardcoded initial state as a placeholder
- make use of dispatch()
- Add and remove books will update books in the store.js and not in the previous component BookLogic.js

### CODE REVIEW

File: src/components/BookItem.js [hunk 1]
```diff
@@ -1,40 +1,48 @@
...
+        <span className="genre">{itemProp.category}</span>
+        <h3 className="title">{itemProp.title}</h3>
+        <span className="author">{itemProp.author}</span>
+        <div className="edit-info">
+          <button type="button">Comments</button>
+          <hr />
+          <button
+            type="button"
+            onClick={() => dispatch(removeBook(itemProp, itemProp.item_id))}
+          >
```
Comment (OyePriscilla) [inline src/components/BookItem.js:19]: - [ ] Please kindly work more on the remove functionality, when I tried to remove a book from the list of newly added books; all the newly added books were removed (though this was not so for the books in the initial state). I guess the same key is assigned for all the childrem. Kindly check your browser console and this [video](https://www.loom.com/share/4c80619613a34f5da386a58e90836efe) for more clarification. 👍

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Diegogagan2587 👍, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
`React component` is wrapped with the <Provider> component from `react-redux` ✔️ 
`Redux store` passed into the <Provider> ✔️ 
Add functionality worked as expected ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookItem.js (hunk 1 above)
Comment (Diegogagan2587) [reply src/components/BookItem.js:19]: Hi @OyePriscilla I really Really really do appreciate your video, thanks, The error was fixed now!

Comment (Diegogagan2587) [reply src/components/BookItem.js:19]: Hi @OyePriscilla I really Really really do appreciate your video, thanks, The error was fixed now!

**Review by DuaneDave — APPROVED** [review]
Hi @Diegogagan2587 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Items are deleted correctly

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**